### PR TITLE
Added HTTPlug support, Guzzle default HTTP Client

### LINKED ISSUE #151: Decouple for Guzzle with HTTPlug
I just started to use your service but I cannot install this library because I have other dependencies require Guzzle 5. 

I would be happy to help decouple from Guzzle and use the HTTPlug abstraction. 

Disclaimer: Im part of the php-http team. We are creator of HTTPlug and maintainers of Guzzle.

### PR DESCRIPTION
**Pull request by Nyholm**
Note, This PR will break BC. 

I removed our `ApiClient` and `RequestBuilder`, without these the API is much easier to read. I also removed src/API/Header since they are not used. 

This will fix #151. 

### Breaking changes

- The 6th constructor argument of `Authentication` was changed to `HttpClient`.
- The 3rd constructor argument of `Management` was changed to `HttpClient`.
- First argument (`$client_id`) of `Auth0\SDK\API\Management::create` was removed. New signature is `Management::create($data)`.

#### Removed classes
- Auth0\SDK\API\Header\Authorization\AuthorizationBearer
- Auth0\SDK\API\Header\ContentType
- Auth0\SDK\API\Header\Header
- Auth0\SDK\API\Helpers\ApiClient
- Auth0\SDK\API\Helpers\RequestBuilder

### Migration

If you are not using the `$guzzleOptions` you are good to go. Now you will have to configure the client yourself. 

```
$token = "eyJhbGciO....eyJhdWQiOiI....1ZVDisdL...";
$domain = "account.auth0.com";
$client = Http\Adapter\Guzzle6::createWithConfig(['verify'=>false, 'timeout'=>2]);

$auth0Api = new Management($token, $domain, client);
```

PS, I found some bugs while implementing this. Some API calls did not use all function arguments.

### CODE REVIEW

**Comment by Nyholm** [comment]
Do we like this approach?

**Comment by Nyholm** [comment]
Can I get any feedback on this before I continue the work?

**Comment by Nyholm** [comment]
Ping?

**Comment by cocojoe** [comment]
Hi @Nyholm 

Thank you for your continued contributions, bandwidth has been a little limited as of late for this repo due to other priorities, so apologies for this.  I'll be helping out and I'll catch up with @glena hopefully today and lets see if we can take a look at these PRs.

Thanks

**Comment by Nyholm** [comment]
Thank you. Im developing a Auth0 specific Symfony bundle. I would love to be able to use the official api client.

**Comment by cocojoe** [comment]
@Nyholm Sorry, I should have said, looks good so far.  That sounds great re Symfony.

**Comment by Nyholm** [comment]
Thank you Martin. 
I will finish up this PR. I just wanted to make sure I do not do tons of work that will go to waste.

File: .env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+GLOBAL_CLIENT_ID=pSm3810KgGx5qnjckjRGoWuBLlU0Hjgl
+GLOBAL_CLIENT_SECRET=OG1G34icGnYQKjtV26Q_qokQe6bafT8K3ecdHwIQ4pcCEiKjfQNBpZkzJdVxg_8s
+DOMAIN=happyr.eu.auth0.com
```
Comment (glena) [inline .env]: try to avoid pushing the credentials. add this to the `.gitignore` and rotate your secret please

Comment (Nyholm) [inline .env]: I must have made a mistake. Thank you. 

(I have a hard time running tests locally)

**Comment by Nyholm** [comment]
Im done with this PR. There is plenty more coming up after this. They will not be this large though..

File: .env (hunk 1 above)
Comment (glena) [inline .env]: yes, mock the api requests is in the backlog but never had time to go through that

**Comment by cocojoe** [comment]
Good work, please can you remove manual comments from the `changelog` these will be generated and during release change.  Also why you want to have a good PR title as this will go into the `changelog` 😄

**Comment by cocojoe** [comment]
@Nyholm can you update the PR comment please and let us know and list any and all breaking changes to the Public API.

Also what code migration is required for the developer?

Thanks

**Comment by cocojoe** [comment]
Please rebase your commits, fixup and reword into something cleaner.

**Comment by Nyholm** [comment]
Thank you for the review. I've squashed my commits, updated the PR comment and removed my entries in changelog.

**Comment by cocojoe** [comment]
@Nyholm once conflict fixed I'll merge this into 6.x.x

**Comment by Nyholm** [comment]
Thank you. I've rebased the PR on 6.x.x

**Comment by Nyholm** [comment]
Awesome. Thank you for merging

**Pull request merged**